src/main/java/UseCases/OtherAccountCases.java

```diff
@@ -27,58 +27,42 @@ public String[] getUserInfo(String userid) {
             userInfo[1] = location;
             userInfo[2] = bio;
 
-            String courseString = "";
+            ArrayList<String> courseString = new ArrayList<>();
             Courses courses = userclass.getCourses();
             if (!courses.hidden) {
                 ArrayList<Course> courseList = courses.data;
-                for (int i = 0; i < courseList.size(); i++) {
-                    Course course = courseList.get(i);
+                for (Course course : courseList) {
                     if (!course.hidden) {
-                        if (courseString.equals("")) {
-                            courseString += ", " + course.data;
-                        }
-                        else {
-                            courseString += course.data;
-                        }
+                        courseString.add(course.data.code);
                     }
                 }
             }
-
-            String interstsString = "";
+            ArrayList<String> interstsList = new ArrayList<>();
             Interests intersts = userclass.getInterests();
             if (!intersts.hidden) {
                 HashMap<Integer, Boolean> map = intersts.data;
                 for (int i = 0; i <= 50; i++) {
```

Comment:
This might not be a good idea to loop through a solid number instead of the attributedict.size()